rmw_fastrtps_shared_cpp/src/qos.cpp

```diff
@@ -21,84 +21,33 @@
 
 #include "rmw/error_handling.h"
 
-bool
-get_datareader_qos(
-  const rmw_qos_profile_t & qos_policies,
-  eprosima::fastrtps::SubscriberAttributes & sattr)
+static
+eprosima::fastrtps::Duration_t
+rmw_time_to_fastrtps(const rmw_time_t & time)
 {
-  switch (qos_policies.history) {
-    case RMW_QOS_POLICY_HISTORY_KEEP_LAST:
-      sattr.topic.historyQos.kind = eprosima::fastrtps::KEEP_LAST_HISTORY_QOS;
-      break;
-    case RMW_QOS_POLICY_HISTORY_KEEP_ALL:
-      sattr.topic.historyQos.kind = eprosima::fastrtps::KEEP_ALL_HISTORY_QOS;
-      break;
-    case RMW_QOS_POLICY_HISTORY_SYSTEM_DEFAULT:
-      break;
-    default:
-      RMW_SET_ERROR_MSG("Unknown QoS history policy");
-      return false;
-  }
-
-  switch (qos_policies.reliability) {
-    case RMW_QOS_POLICY_RELIABILITY_BEST_EFFORT:
-      sattr.qos.m_reliability.kind = eprosima::fastrtps::BEST_EFFORT_RELIABILITY_QOS;
-      break;
-    case RMW_QOS_POLICY_RELIABILITY_RELIABLE:
-      sattr.qos.m_reliability.kind = eprosima::fastrtps::RELIABLE_RELIABILITY_QOS;
-      break;
-    case RMW_QOS_POLICY_RELIABILITY_SYSTEM_DEFAULT:
-      break;
-    default:
-      RMW_SET_ERROR_MSG("Unknown QoS reliability policy");
-      return false;
-  }
-
-  switch (qos_policies.durability) {
-    case RMW_QOS_POLICY_DURABILITY_TRANSIENT_LOCAL:
-      sattr.qos.m_durability.kind = eprosima::fastrtps::TRANSIENT_LOCAL_DURABILITY_QOS;
-      break;
-    case RMW_QOS_POLICY_DURABILITY_VOLATILE:
-      sattr.qos.m_durability.kind = eprosima::fastrtps::VOLATILE_DURABILITY_QOS;
-      break;
-    case RMW_QOS_POLICY_DURABILITY_SYSTEM_DEFAULT:
-      break;
-    default:
-      RMW_SET_ERROR_MSG("Unknown QoS durability policy");
-      return false;
-  }
-
-  if (qos_policies.depth != RMW_QOS_POLICY_DEPTH_SYSTEM_DEFAULT) {
-    sattr.topic.historyQos.depth = static_cast<int32_t>(qos_policies.depth);
-  }
-
-  // ensure the history depth is at least the requested queue size
-  assert(sattr.topic.historyQos.depth >= 0);
-  if (
-    eprosima::fastrtps::KEEP_LAST_HISTORY_QOS == sattr.topic.historyQos.kind &&
-    static_cast<size_t>(sattr.topic.historyQos.depth) < qos_policies.depth)
-  {
-    if (qos_policies.depth > (std::numeric_limits<int32_t>::max)()) {
-      RMW_SET_ERROR_MSG(
-        "failed to set history depth since the requested queue size exceeds the DDS type");
-      return false;
-    }
-    sattr.topic.historyQos.depth = static_cast<int32_t>(qos_policies.depth);
-  }
-
-  return true;
+  return eprosima::fastrtps::Duration_t(time.sec, time.nsec);
```

Comment:
This line still results in compiler warnings on Windows.

Reply:
Fixed.